### PR TITLE
Switch to ruff and port to mesonpy

### PR DESCRIPTION
**Pull request by alexfikl**
This does a bit of modernizing as the title says.

The examples are pretty out of date, but the rest of the things seems to be working.

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+build-backend = "mesonpy"
+requires = [
+    "meson-python",
+    "numpy",
+    "wheel",
+    "pybind11",
+]
+
+[project]
+name = "meshpy"
```
Comment (alexfikl) [inline pyproject.toml:11]: Hm, I think the project name was `MeshPy` before. Should rename it back?

**Comment by alexfikl** [comment]
I've looked around a bit and think I know why the wheel build failures are happening.

1. On Windows, it seems like it's a 32-bit vs 64-bit issue: trying to build a 32 bit extension
with a 64 bit compiler. Numpy had the same problem https://github.com/numpy/numpy/issues/24704.

2. On Ubuntu, it's an issue with the available numpy wheels. They don't have wheels for pypy38, pypy310 and some i686 combinations in there. That makes the CI try to build numpy from scratch and (besides taking a long time) it eventually fails on `PyPy 3.9 manylinux i686`.

File: pyproject.toml (hunk 1 above)
Comment (inducer) [reply pyproject.toml:11]: These things are not case-sensitive, so :shrug:. :)

File: meson.build
Comment (inducer) [inline meson.build:1]: You're trying to make sure that I have to know *all* the build tools. I appreciate the sentiment. :facepalm: :grin:

**Comment by inducer** [comment]
> EDIT: I've added additional skips `pp*-manylinux_i686 *-win32`. Let me know if you want to support any of those.

Nah. (But thanks for asking.) My logic is mostly, support $\Leftrightarrow$ (mainstream $\lor$ easy).

**Comment by inducer** [comment]
Didn't see anything objectionable, so :rocket: let's do it.

**Comment by inducer** [comment]
Thanks!

File: meson.build
Comment (alexfikl) [reply meson.build:1]: Haha, sorry about that! I was just being lazy and started copying from `pyfmmlib` :(

I also thought that they used a different editable machinery that allowed pylint to work.. but apparently not :\

**Pull request merged**